microsoft/atlas-design
# Add component governance task

Task: task-[452746](https://dev.azure.com/ceapex/Engineering/_boards/board/t/Fox%20Team/Stories/?workitem=452746)

Link: [preview-225](https://design.docs.microsoft.com/pulls/225)

This PR adds a component governance task to the Azure pipeline build so that it will raise alerts for any package vulnerabilities for each PR. Note that it will not block PR completion.

I also ran `npm audit fix --legacy-peer-deps` for bumping `trim-newlines` to `3.0.1` to fix this [vulnerability](https://dev.azure.com/ceapex/Engineering/_componentGovernance/180005/alert/5098820?typeId=6760326) . The risk of  regular expression denial-of-service is low, but it seems like it can be upgraded without any adverse effects.

## Testing

1. Review the changes.
2. Visit the link above and make sure everything still works as expected.
3. `trim-newlines` is a dependency for `@changesets/cli`, so make and commit a small change in one of the pages. Then, run `npx changeset` to test that you can still make a changeset.
4. Visit https://dev.azure.com/ceapex/Engineering/_build/results?buildId=508327&view=results to see that the vulnerability for `trim-newlines 3.0.0` no longer exists.
